### ISSUE TITLE
Merge params from difference source

### ISSUE THREAD

**Issue by kevinresol** [haxetink/tink_web#75]
I am attempting to write API for AWS. Simply put, they put params everywhere (path,query,header,body).

I would like to hide that complexity from the users by merging them into a single object:

```haxe
@:get('/${opt.foo}')
@:params(opt.bar('bar') in query, opt.baz('x-aws-baz') in header)
function get(opt:{foo:Int, bar:String, baz:String}):Promise<{}>;
```

**Comment by kevinresol** [comment]
First of all we need some consensus on the meta syntax, @back2dos any comments?

**Comment by back2dos** [comment]
Hmmmm. Ok, seems there are two things here:

1. aliasing
2. merging

As for aliasing, I suppose the call syntax is fine. Personally, I would go with `param = query['alias']`. And yeah, merging can be facilitated by allowing the lhs of the `in` or `=` rule be more complex. Meaning the above would be:

```haxe
@:params(opt.bar = query['bar'], opt.baz = header['x-aws-baz'])
// if so preferred, can be shortened to 
@:params(opt.bar in query, opt.baz = header['x-aws-baz'])
```

**Comment by kevinresol** [comment]
OK I just spent a whole afternoon on this and seems it is more complex than I expected. I think we need some serious housekeeping work here. So many edge cases everywhere....

**Comment by kevinresol** [comment]
Working branch: https://github.com/haxetink/tink_web/tree/merge_params

**Comment by kevinresol** [comment]
Here is a list of the supported syntaxes I implemented in #86 

```haxe
@:params(<ident> in <query|header|body>)
@:params(<ident> = <query|header|body>)
@:params(<ident> = <query|header|body>["native"])
@:params(<ident.field> in <query|header|body>)
@:params(<ident.field> = <query|header|body>["native"])
```

Are they complete? or any comments?

**Comment by back2dos** [comment]
Well, nothing is ever complete, but this definitely looks good to me ;)
